In [135]:
import json
import pandas as pd

In [136]:
data = pd.read_json('sizes.json', lines=True)


In [137]:
data.head()

__v                                   _id  \
0    0  {'$oid': '557abe8565950003006fefe7'}   
1    0  {'$oid': '557abe8565950003006fefe5'}   
2    0  {'$oid': '557abe8565950003006fefe8'}   
3    0  {'$oid': '557abe8565950003006fefe6'}   
4    0  {'$oid': '557abe8565950003006fefe9'}   

                              created_at filter_label label  old_label  \
0  {'$date': '2015-06-12T11:12:05.248Z'}            S    26      [26\]   
1  {'$date': '2015-06-12T11:12:05.244Z'}           XS    24  [24, 24\]   
2  {'$date': '2015-06-12T11:12:05.249Z'}            S    28      [28\]   
3  {'$date': '2015-06-12T11:12:05.247Z'}            S    25       [25]   
4  {'$date': '2015-06-12T11:12:05.249Z'}            M    30      [30\]   

                              updated_at  
0  {'$date': '2015-07-07T16:29:11.868Z'}  
1  {'$date': '2015-07-07T16:27:03.743Z'}  
2  {'$date': '2015-07-07T16:36:22.139Z'}  
3  {'$date': '2015-07-07T16:29:03.370Z'}  
4  {'$date': '2015-07-07T16:44:25.382Z'}

In [138]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51210 entries, 0 to 51209
Data columns (total 7 columns):
__v             51210 non-null int64
_id             51210 non-null object
created_at      51210 non-null object
filter_label    294 non-null object
label           51210 non-null object
old_label       51210 non-null object
updated_at      51210 non-null object
dtypes: int64(1), object(6)
memory usage: 2.7+ MB


**Observations**
- dataframe has 7 columns/ 51210 lines
- filter_label column has only 294 entries, vs 51 210 lines
- dataframe entries are objects which is not the most suitable type for data manipulation

**Data frame columns - detailed view**

In [139]:
column_names=list(data)

In [140]:
column_types=[type(data._id[0]),
type(data.__v[0]),
type(data.created_at[0]),
type(data.filter_label[0]),
type(data.label[0]),
type(data.old_label[0]),
type(data.updated_at[0])]

In [141]:
for i in range (0,len(column_names)):
    print(column_names[i],column_types[i])

__v <class 'dict'>
_id <class 'numpy.int64'>
created_at <class 'dict'>
filter_label <class 'str'>
label <class 'str'>
old_label <class 'list'>
updated_at <class 'dict'>


** Step 1 - flatten the dataframe** 

1.1 Column **_if**: explore dictionary keys and values

In [142]:
col_name='_id'

In [143]:
def flatten_dic(df,col_name):
    keys=[]
    values=[]
    for index, row in data.iterrows():
        item=row[col_name]
        keys.extend(item.keys())
        values.extend(item.values())
    return keys, values

In [144]:
keys_id, values_id= flatten_dic(data, col_name)

In [145]:
def check_unique(item):
    unq_items=set(item)
    return len(unq_items)

In [146]:
print(check_unique(keys_id))

1


In [147]:
print(check_unique(values_id))

51210


**Observations**
- The keys are the same across all lines, therefore, can be removed, and only ID value can remain as item ID identifier

In [148]:
def clean_col(data,col_name):
    for index,row in data.iterrows():
        item=row[col_name]
        value=list(item.values())    
        data.loc[data.index[index], col_name]=str(value[0])
    return data
 


In [ ]:
clean_col(data, '_id')

In [150]:
data.head()

__v                       _id                             created_at  \
0    0  557abe8565950003006fefe7  {'$date': '2015-06-12T11:12:05.248Z'}   
1    0  557abe8565950003006fefe5  {'$date': '2015-06-12T11:12:05.244Z'}   
2    0  557abe8565950003006fefe8  {'$date': '2015-06-12T11:12:05.249Z'}   
3    0  557abe8565950003006fefe6  {'$date': '2015-06-12T11:12:05.247Z'}   
4    0  557abe8565950003006fefe9  {'$date': '2015-06-12T11:12:05.249Z'}   

  filter_label label  old_label                             updated_at  
0            S    26      [26\]  {'$date': '2015-07-07T16:29:11.868Z'}  
1           XS    24  [24, 24\]  {'$date': '2015-07-07T16:27:03.743Z'}  
2            S    28      [28\]  {'$date': '2015-07-07T16:36:22.139Z'}  
3            S    25       [25]  {'$date': '2015-07-07T16:29:03.370Z'}  
4            M    30      [30\]  {'$date': '2015-07-07T16:44:25.382Z'}

In [151]:
col_name2='created_at'

In [152]:
keys_cr_at, values_cr_at=flatten_dic(data,col_name2)

In [153]:
print(check_unique(keys_cr_at))

1


In [154]:
print(check_unique(values_cr_at))

43535


In [155]:
### observation: there are 43 535 craeted_at values, less than lines in the DF, therefore, we can assume this column has nan values 

In [ ]:
clean_col(data, col_name2)

In [157]:
data.head()

__v                       _id                created_at filter_label label  \
0    0  557abe8565950003006fefe7  2015-06-12T11:12:05.248Z            S    26   
1    0  557abe8565950003006fefe5  2015-06-12T11:12:05.244Z           XS    24   
2    0  557abe8565950003006fefe8  2015-06-12T11:12:05.249Z            S    28   
3    0  557abe8565950003006fefe6  2015-06-12T11:12:05.247Z            S    25   
4    0  557abe8565950003006fefe9  2015-06-12T11:12:05.249Z            M    30   

   old_label                             updated_at  
0      [26\]  {'$date': '2015-07-07T16:29:11.868Z'}  
1  [24, 24\]  {'$date': '2015-07-07T16:27:03.743Z'}  
2      [28\]  {'$date': '2015-07-07T16:36:22.139Z'}  
3       [25]  {'$date': '2015-07-07T16:29:03.370Z'}  
4      [30\]  {'$date': '2015-07-07T16:44:25.382Z'}

In [158]:
data['created_at']  = pd.to_datetime(data['created_at'])

In [159]:
data.head()

__v                       _id              created_at filter_label label  \
0    0  557abe8565950003006fefe7 2015-06-12 11:12:05.248            S    26   
1    0  557abe8565950003006fefe5 2015-06-12 11:12:05.244           XS    24   
2    0  557abe8565950003006fefe8 2015-06-12 11:12:05.249            S    28   
3    0  557abe8565950003006fefe6 2015-06-12 11:12:05.247            S    25   
4    0  557abe8565950003006fefe9 2015-06-12 11:12:05.249            M    30   

   old_label                             updated_at  
0      [26\]  {'$date': '2015-07-07T16:29:11.868Z'}  
1  [24, 24\]  {'$date': '2015-07-07T16:27:03.743Z'}  
2      [28\]  {'$date': '2015-07-07T16:36:22.139Z'}  
3       [25]  {'$date': '2015-07-07T16:29:03.370Z'}  
4      [30\]  {'$date': '2015-07-07T16:44:25.382Z'}

In [160]:
col_name3='updated_at'

In [161]:
keys_updt_at, values_updt_at=flatten_dic(data,col_name3)

In [162]:
print(check_unique(keys_updt_at))

1


In [163]:
print(check_unique(values_updt_at))

46594


In [164]:
### observation: there are 46 594 craeted_at values, less than lines in the DF, therefore, we can assume this column has nan values 

In [165]:
clean_col(data, col_name3)

__v                       _id              created_at filter_label  \
0        0  557abe8565950003006fefe7 2015-06-12 11:12:05.248            S   
1        0  557abe8565950003006fefe5 2015-06-12 11:12:05.244           XS   
2        0  557abe8565950003006fefe8 2015-06-12 11:12:05.249            S   
3        0  557abe8565950003006fefe6 2015-06-12 11:12:05.247            S   
4        0  557abe8565950003006fefe9 2015-06-12 11:12:05.249            M   
5        0  557abe8565950003006ff021 2015-06-12 11:12:05.269    0-1 years   
6        0  557abe8565950003006ff025 2015-06-12 11:12:05.270    0-1 years   
7        0  557abe8565950003006ff033 2015-06-12 11:12:05.275    1-2 years   
8        0  557abe8565950003006feffa 2015-06-12 11:12:05.258          NaN   
9        0  557abe8565950003006feff7 2015-06-12 11:12:05.257          NaN   
10       0  557abe8565950003006ff036 2015-06-12 11:12:05.276    1-2 years   
11       0  557abe8565950003006ff000 2015-06-12 11:12:05.259          NaN   
12       0  557abe8565950003006ff050 2015-06-12 11:12:05.288          NaN   
13       0  557abe8565950003006feffd 2015-06-12 11:12:05.259          NaN   
14       0  557abe8565950003006ff001 2015-06-12 11:12:05.260          NaN   
15       0  557abe8565950003006ff065 2015-06-12 11:12:05.295          NaN   
16       0  557abe8565950003006ff00f 2015-06-12 11:12:05.265          NaN   
17       0  557abe8565950003006ff004 2015-06-12 11:12:05.260          NaN   
18       0  557abe8565950003006ff020 2015-06-12 11:12:05.268    0-1 years   
19       0  557abe8565950003006ff030 2015-06-12 11:12:05.274    1-2 years   
20       0  557abe8565950003006ff078 2015-06-12 11:12:05.305    5-7 years   
21       0  557abe8565950003006ff013 2015-06-12 11:12:05.266          NaN   
22       0  557abe8565950003006ff022 2015-06-12 11:12:05.269    1-2 years   
23       0  557abe8565950003006ff032 2015-06-12 11:12:05.275    0-1 years   
24       0  557abe8565950003006ff081 2015-06-12 11:12:05.307   8-10 years   
25       0  557abe8565950003006ff029 2015-06-12 11:12:05.272    2-4 years   
26       0  557abe8565950003006ff031 2015-06-12 11:12:05.274    1-2 years   
27       0  557abe8565950003006ff06c 2015-06-12 11:12:05.303          NaN   
28       0  557abe8565950003006ff084 2015-06-12 11:12:05.308   8-10 years   
29       0  557abe8565950003006ff02f 2015-06-12 11:12:05.273    0-1 years   
...    ...                       ...                     ...          ...   
51180    0  5b2bfa10df49450400fd53ff 2018-06-21 19:18:40.833          NaN   
51181    0  5b2bfa10df49450400fd5400 2018-06-21 19:18:40.852          NaN   
51182    0  5b2bfa10df49450400fd5401 2018-06-21 19:18:40.852          NaN   
51183    0  5b2bfa56df49450400fd5479 2018-06-21 19:19:50.169          NaN   
51184    0  5b2c02433ae91c04002e56c1 2018-06-21 19:53:39.990          NaN   
51185    0  5b2c639a5c69fa04005dcea7 2018-06-22 02:48:58.537          NaN   
51186    0  5b2c8aa0b826e504001c7fea 2018-06-22 05:35:28.448          NaN   
51187    0  5b2c8aa0b826e504001c7feb 2018-06-22 05:35:28.451          NaN   
51188    0  5b2c8ab323f1b504003f10e3 2018-06-22 05:35:47.692          NaN   
51189    0  5b2c8ab323f1b504003f10e4 2018-06-22 05:35:47.696          NaN   
51190    0  5b2c8c262d60830400bd1d4b 2018-06-22 05:41:58.769          NaN   
51191    0  5b2c8d1a23f1b504003f14d7 2018-06-22 05:46:02.206          NaN   
51192    0  5b2c8d50af13b60400841f27 2018-06-22 05:46:56.524          NaN   
51193    0  5b2c8d94af13b60400841f75 2018-06-22 05:48:04.464          NaN   
51194    0  5b2c8e0a5c69fa04005dee67 2018-06-22 05:50:02.280          NaN   
51195    0  5b2c8e102d60830400bd203f 2018-06-22 05:50:08.129          NaN   
51196    0  5b2c8e7e53099704006de2b6 2018-06-22 05:51:58.136          NaN   
51197    0  5b2c92e223f1b504003f1c5f 2018-06-22 06:10:42.341          NaN   
51198    0  5b2c92f35c69fa04005df4df 2018-06-22 06:10:59.751          NaN   
51199    0  5b2c92f8bff5e50400d5b890 2018-06-22 06:11:04.922          NaN   
512

In [166]:
data['updated_at']  = pd.to_datetime(data['updated_at'])

In [167]:
data.head()

__v                       _id              created_at filter_label label  \
0    0  557abe8565950003006fefe7 2015-06-12 11:12:05.248            S    26   
1    0  557abe8565950003006fefe5 2015-06-12 11:12:05.244           XS    24   
2    0  557abe8565950003006fefe8 2015-06-12 11:12:05.249            S    28   
3    0  557abe8565950003006fefe6 2015-06-12 11:12:05.247            S    25   
4    0  557abe8565950003006fefe9 2015-06-12 11:12:05.249            M    30   

   old_label              updated_at  
0      [26\] 2015-07-07 16:29:11.868  
1  [24, 24\] 2015-07-07 16:27:03.743  
2      [28\] 2015-07-07 16:36:22.139  
3       [25] 2015-07-07 16:29:03.370  
4      [30\] 2015-07-07 16:44:25.382

In [168]:
#We observe, that each cell contains a list of items.
#Convert the list into a string, separated by ','

In [169]:
for index,row in data.iterrows():
    item=row['old_label']
    item=', '.join(item)
    item=re.sub('[^A-Za-z0-9,-]+', ' ', item)
    data.loc[data.index[index], 'old_label']=item


In [173]:
data.head()

__v                       _id              created_at filter_label label  \
0    0  557abe8565950003006fefe7 2015-06-12 11:12:05.248            S    26   
1    0  557abe8565950003006fefe5 2015-06-12 11:12:05.244           XS    24   
2    0  557abe8565950003006fefe8 2015-06-12 11:12:05.249            S    28   
3    0  557abe8565950003006fefe6 2015-06-12 11:12:05.247            S    25   
4    0  557abe8565950003006fefe9 2015-06-12 11:12:05.249            M    30   

  old_label              updated_at  
0       26  2015-07-07 16:29:11.868  
1   24, 24  2015-07-07 16:27:03.743  
2       28  2015-07-07 16:36:22.139  
3        25 2015-07-07 16:29:03.370  
4       30  2015-07-07 16:44:25.382

In [175]:
#identify if cell contains numbers only, strings only, combination of both or, is neither: 
#for instance, some strings contain both: a number and a string
#assign each cell to a category
#1: strings only
#2: numbers only
#3 strings and numbers
#4 mixture of all 

In [193]:
datatest=data.loc[:20]

In [194]:
datatest

__v                       _id              created_at filter_label  \
0     0  557abe8565950003006fefe7 2015-06-12 11:12:05.248            S   
1     0  557abe8565950003006fefe5 2015-06-12 11:12:05.244           XS   
2     0  557abe8565950003006fefe8 2015-06-12 11:12:05.249            S   
3     0  557abe8565950003006fefe6 2015-06-12 11:12:05.247            S   
4     0  557abe8565950003006fefe9 2015-06-12 11:12:05.249            M   
5     0  557abe8565950003006ff021 2015-06-12 11:12:05.269    0-1 years   
6     0  557abe8565950003006ff025 2015-06-12 11:12:05.270    0-1 years   
7     0  557abe8565950003006ff033 2015-06-12 11:12:05.275    1-2 years   
8     0  557abe8565950003006feffa 2015-06-12 11:12:05.258          NaN   
9     0  557abe8565950003006feff7 2015-06-12 11:12:05.257          NaN   
10    0  557abe8565950003006ff036 2015-06-12 11:12:05.276    1-2 years   
11    0  557abe8565950003006ff000 2015-06-12 11:12:05.259          NaN   
12    0  557abe8565950003006ff050 2015-06-12 11:12:05.288          NaN   
13    0  557abe8565950003006feffd 2015-06-12 11:12:05.259          NaN   
14    0  557abe8565950003006ff001 2015-06-12 11:12:05.260          NaN   
15    0  557abe8565950003006ff065 2015-06-12 11:12:05.295          NaN   
16    0  557abe8565950003006ff00f 2015-06-12 11:12:05.265          NaN   
17    0  557abe8565950003006ff004 2015-06-12 11:12:05.260          NaN   
18    0  557abe8565950003006ff020 2015-06-12 11:12:05.268    0-1 years   
19    0  557abe8565950003006ff030 2015-06-12 11:12:05.274    1-2 years   
20    0  557abe8565950003006ff078 2015-06-12 11:12:05.305    5-7 years   

             label                                          old_label  \
0               26                                                26    
1               24                                            24, 24    
2               28                                                28    
3               25                                                 25   
4               30                                                30    
5    6 - 12 months  6 - 12m, 6-12 months, 6-12m, 6- 12 Months, 6 -...   
6     3 - 6 months  3 - 6 months, 3 - 6m, 3-6 Months, 3-6m, EU 16 ...   
7          2 years  2yrs, 2 years, 2y, Age 2, 2 years 92 , 2 years...   
8               54                                                 54   
9               48                                                 48   
10  12 - 24 months                                       12-24 months   
11               C                                                  C   
12            Sand                                               sand   
13            Pink  Pink, fluorescent pink, pink grapefruit, Pink ...   
14               D                                                  D   
15      Anthracite                                         Anthracite   
16               N                     N, UK N, Size N, 53, Large N-O   
17   Full alphabet                                      Full alphabet   
18  12 - 18 months  12-18m, 12 - 18m, 12 - 18 Months, 12-18 months...   
19  18 - 24 months  18-24m, 18 - 24m, 18 - 24 Months, 18-24, 18-24...   
20         6 years   6 years, Age 6, 6y, 6 years 116 , 6 years 116cm    

                updated_at  
0  2015-07-07 16:29:11.868  
1  2015-07-07 16:27:03.743  
2  2015-07-07 16:36:22.139  
3  2015-07-07 16:29:03.370  
4  2015-07-07 16:44:25.382  
5  2015-07-07 16:19:16.195  
6  2015-07-07 16:12:48.379  
7  2015-07-07 16:12:06.255  
8  2015-06-12 11:12:05.457  
9  2015-06-12 11:12:05.456  
10 2015-07-07 16:08:48.845  
11 2015-06-12 11:12:05.463  
12 2015-06-12 11:12:05.560  
13 2015-06-12 11:12:05.459  
14 2015-06-12 11:12:05.463  
15 2015-06-12 11:12:05.576  
16 2015-06-12 11:12:05.504  
17 2015-06-12 11:12:05.466  
18 2015-07-07 16:08:40.072  
19 2015-07-07 16:09:43.702  
20 2015-07-07 16:21:09.991

In [213]:
for index,row in datatest.iterrows():
    item=row['old_label']
    print(type(item))

<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>


In [258]:
for index,row in datatest.iterrows():
    item=row['old_label']
    label=[]
    #result = [x.strip() for x in item.split(',')]
    result=item.replace(' ','').split(',')
    for r in result:
        if r.isdigit():
            #print('number',r)
            label.append('1')
        elif r.isalpha():
            #print('letters',r)
            label.extend('2')
        else:
            #print('misc',r)
            label.extend('3')
    final_label=set(label)
    if len(final_label)==1:
        info_type=list(final_label)[0]
        print(info_type)
        datatest['info_type']=info_type
    else:
        info_type==['3']
        datatest['info_type']=info_type
        
    




1
1

C:\Users\Gabi\Anaconda3\lib\site-packages\ipykernel_launcher.py:20: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy



1
1
1
3
3
3
1
1
3
2
2
2
2
2
2
3
3
3


C:\Users\Gabi\Anaconda3\lib\site-packages\ipykernel_launcher.py:23: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


In [257]:
datatest

__v                       _id              created_at filter_label  \
0     0  557abe8565950003006fefe7 2015-06-12 11:12:05.248            S   
1     0  557abe8565950003006fefe5 2015-06-12 11:12:05.244           XS   
2     0  557abe8565950003006fefe8 2015-06-12 11:12:05.249            S   
3     0  557abe8565950003006fefe6 2015-06-12 11:12:05.247            S   
4     0  557abe8565950003006fefe9 2015-06-12 11:12:05.249            M   
5     0  557abe8565950003006ff021 2015-06-12 11:12:05.269    0-1 years   
6     0  557abe8565950003006ff025 2015-06-12 11:12:05.270    0-1 years   
7     0  557abe8565950003006ff033 2015-06-12 11:12:05.275    1-2 years   
8     0  557abe8565950003006feffa 2015-06-12 11:12:05.258          NaN   
9     0  557abe8565950003006feff7 2015-06-12 11:12:05.257          NaN   
10    0  557abe8565950003006ff036 2015-06-12 11:12:05.276    1-2 years   
11    0  557abe8565950003006ff000 2015-06-12 11:12:05.259          NaN   
12    0  557abe8565950003006ff050 2015-06-12 11:12:05.288          NaN   
13    0  557abe8565950003006feffd 2015-06-12 11:12:05.259          NaN   
14    0  557abe8565950003006ff001 2015-06-12 11:12:05.260          NaN   
15    0  557abe8565950003006ff065 2015-06-12 11:12:05.295          NaN   
16    0  557abe8565950003006ff00f 2015-06-12 11:12:05.265          NaN   
17    0  557abe8565950003006ff004 2015-06-12 11:12:05.260          NaN   
18    0  557abe8565950003006ff020 2015-06-12 11:12:05.268    0-1 years   
19    0  557abe8565950003006ff030 2015-06-12 11:12:05.274    1-2 years   
20    0  557abe8565950003006ff078 2015-06-12 11:12:05.305    5-7 years   

             label                                          old_label  \
0               26                                                26    
1               24                                            24, 24    
2               28                                                28    
3               25                                                 25   
4               30                                                30    
5    6 - 12 months  6 - 12m, 6-12 months, 6-12m, 6- 12 Months, 6 -...   
6     3 - 6 months  3 - 6 months, 3 - 6m, 3-6 Months, 3-6m, EU 16 ...   
7          2 years  2yrs, 2 years, 2y, Age 2, 2 years 92 , 2 years...   
8               54                                                 54   
9               48                                                 48   
10  12 - 24 months                                       12-24 months   
11               C                                                  C   
12            Sand                                               sand   
13            Pink  Pink, fluorescent pink, pink grapefruit, Pink ...   
14               D                                                  D   
15      Anthracite                                         Anthracite   
16               N                     N, UK N, Size N, 53, Large N-O   
17   Full alphabet                                      Full alphabet   
18  12 - 18 months  12-18m, 12 - 18m, 12 - 18 Months, 12-18 months...   
19  18 - 24 months  18-24m, 18 - 24m, 18 - 24 Months, 18-24, 18-24...   
20         6 years   6 years, Age 6, 6y, 6 years 116 , 6 years 116cm    

                updated_at info_type  
0  2015-07-07 16:29:11.868         3  
1  2015-07-07 16:27:03.743         3  
2  2015-07-07 16:36:22.139         3  
3  2015-07-07 16:29:03.370         3  
4  2015-07-07 16:44:25.382         3  
5  2015-07-07 16:19:16.195         3  
6  2015-07-07 16:12:48.379         3  
7  2015-07-07 16:12:06.255         3  
8  2015-06-12 11:12:05.457         3  
9  2015-06-12 11:12:05.456         3  
10 2015-07-07 16:08:48.845         3  
11 2015-06-12 11:12:05.463         3  
12 2015-06-12 11:12:05.560         3  
13 2015-06-12 11:12:05.459         3  
14 2015-06-12 11:12:05.463         3  
15 2015-06-12 11:12:05.576         3  
16 2015-06-12 11:12:05.504         3  
17 2015-06-12 11:12:05.466         3  
18 2015-07-07 16:08:40.072 

In [ ]:
    if len(final_label)==1:
        print((final_label)[0])
           
        datatest['info_type'] = list(final_label)[0]
    else:
        datatest['info_type']=='3'
    #print(label)
 
 

In [242]:
datatest.head()

__v                       _id              created_at filter_label label  \
0    0  557abe8565950003006fefe7 2015-06-12 11:12:05.248            S    26   
1    0  557abe8565950003006fefe5 2015-06-12 11:12:05.244           XS    24   
2    0  557abe8565950003006fefe8 2015-06-12 11:12:05.249            S    28   
3    0  557abe8565950003006fefe6 2015-06-12 11:12:05.247            S    25   
4    0  557abe8565950003006fefe9 2015-06-12 11:12:05.249            M    30   

  old_label              updated_at info_type  
0       26  2015-07-07 16:29:11.868         3  
1   24, 24  2015-07-07 16:27:03.743         3  
2       28  2015-07-07 16:36:22.139         3  
3        25 2015-07-07 16:29:03.370         3  
4       30  2015-07-07 16:44:25.382         3

In [226]:
fooo=datatest['old_label'].iloc[13]

In [228]:
fooo

'Pink, fluorescent pink, pink grapefruit, Pink Flag'

In [229]:
fooo1=[x.strip() for x in fooo.split(',')]

In [230]:
fooo1

['Pink', 'fluorescent pink', 'pink grapefruit', 'Pink Flag']

In [231]:
for f1 in fooo1:
    if f1.isalpha():
        print('yes')
    else:
        print('no')
    

yes
no
no
no


In [179]:
d='2324'


In [180]:
if d.isdigit():
    print('yes')
else:
    print('no')

yes


In [191]:
s='foo'

In [227]:
if fooo.isalpha():
    print('yes')
else:
    print('no')

no
